src/Toml/Parser.hs

```diff
@@ -193,10 +261,10 @@ arrayP = lexeme $ between (char '[' *> sc) (char ']') elements
 
 valueP :: Parser UValue
 valueP = UBool    <$> boolP
+     <|> UDate    <$> dateTimeP
```

Comment:
Since `dateTimeP` parser is going to be very slow, it's better to move it at the end of `valueP` function.

Reply:
If `integerP` is tried before `dateTimeP` it will parse a part of the date as an integer, such in this case: "12:24:14" (`integerP` will parse `12`).

I was thinking that `doubleP` should also be tried before `dateTimeP`, but I can't think of such case now.

So it's possible to move `dateTimeP` to the end of `valueP` as long as `integerP` is after it.

Reply:
@ghallak Hmm. that's a problem indeed. Okay, let's have dates earlier. Maybe we will figure out something later.